quickwit/quickwit-indexing/src/models/indexing_service_message.rs

```diff
@@ -52,7 +52,15 @@ pub struct ShutdownPipeline {
 /// server. This is mostly useful for ad-hoc indexing pipelines launched with `quickwit index
 /// ingest ..` and testing.
 #[derive(Debug)]
-pub struct DetachPipeline {
+pub struct DetachIndexingPipeline {
+    pub pipeline_id: IndexingPipelineId,
+}
+
+#[derive(Debug)]
+/// Detaches a merge pipeline from the indexing service. The pipeline is no longer managed by the
+/// server. This is mostly useful for preventing the server killing an existing merge pipeline
+/// if a indexing pipeline is detached.
+pub struct DetachMergePipeline {
     pub pipeline_id: IndexingPipelineId,
```

Comment:
This should really be `MergePipelineId` but in other applications like spawning a merge pipeline, we use an `IndexingPipelineId` so for now in this WIP I've left it as an `IndexingPipelineId` which then gets converted to a `MergePipelineId`

Reply:
Ok, this is weird indeed, the spawn merge pipeline seems to be related to the way we used to spawn pipelines: one merge pipeline for one indexing pipeline.

If we look at `spawn_merge_pipeline` function, it is calling `spawn_pipeline_inner`... so really this is not great. Can you just remove the `spawn_merge_pipeline` and `SpawnMergePipeline` which is useless?

Reply:
And then you can really detach a merge pipeline with a `MergePipelineId`.

Reply:
Sounds like a plan, I noticed we do something equally weird as part of the merge CLI where we create a merge pipeline, then detach the resultant indexing pipeline, which we should probably cleanup as part of this.

Reply:
Yes the merge CLI is also somewhat broken. A plan for the merge CLI would be:
- start the indexing pipeline
- detach the merge pipeline (I'm not sure we need to detach the indexing pipeline)
- use observe command to get the `MergePlannerState` (it's easy to get a given actor from the universe and send him the observe message). If the `state.ongoing_merge_operations` are not empty, wait a 1 second, observe again, etc.

We can do that in a separated PR.

Reply:
I've gone through and cleaned all this up now 👍 The merge CLI has been slightly adjusted to use `spawn_pipeline` so we can remove spawn_merge_pipeline.